spec/features/users/sign_up_spec.rb

```diff
@@ -23,6 +23,21 @@
     end
   end
 
+  context 'user cancels on the enter password screen' do
+    it 'deletes the user record and returns them to the home page' do
+      email = 'test@test.com'
+
+      visit sign_up_email_path
+
+      submit_form_with_valid_email
+      click_confirmation_link_in_email(email)
+      click_on t('links.cancel_account_creation')
+
+      expect(current_path).to eq root_path
+      expect(User.find_with_email(email)).to be_nil
```

Comment:
My 2c: by convention I try to avoid doing things in feature specs that a user could not do. That doesn't mean I don't cheat occasionally, such as in the IdV process where I check the status of a record or that encryption worked successfully. But I try and avoid that, and prefer a request spec instead if I need to.